server/lib/oauth2/oauth2.executeQuery.js

```diff
@@ -0,0 +1,136 @@
+const queryString = require('query-string');
+const { ClientCredentials } = require('simple-oauth2');
+const { default: axios } = require('axios');
+const logger = require('../../utils/logger');
+const providers = require('../../config/oauth2/providers.json');
+
+/**
+ * @description Refresh Oauth2 token in case of it expired .
+ * @param {string} serviceId - Gladys service id call method.
+ * @param {string} integrationName - Name of oauth2 integration.
+ * @param {Object} gladys - Current gladys instance.
+ * @param {string} userId - Gladys userId of current session.
+ * @returns {Object} Response of oauth2 refresh token query .
+ * @example
+ * oauth2.refreshTokenAccess('7fdsf4s68r4gfr68f4r63csd7f6f4c3r85',
+ *    'withings', {...}, '78v4f3df83g74v1fsd8375f63gvrf5c');
+ */
+async function refreshTokenAccess(serviceId, integrationName, gladys, userId) {
+  const clientId = await gladys.variable.getValue(`${integrationName.toUpperCase()}_CLIENT_ID`, serviceId, userId);
+  const secretId = await gladys.variable.getValue(`${integrationName.toUpperCase()}_SECRET_ID`, serviceId, userId);
+  const accessToken = JSON.parse(
+    await gladys.variable.getValue(`${integrationName.toUpperCase()}_ACCESS_TOKEN`, serviceId, userId),
+  );
+
+  logger.trace('accessToken: ', accessToken);
+  // Find provider configuration
+  const currentProvider = providers[integrationName];
```

Comment:
Only test? According to providers.json.
I think required values should be stored in DB by service.

Reply:
I remove provider.json to put it in db